### PR TITLE
[Redesign] Improved Form Creation

### PR DESCRIPTION
**Pull request by loic-sharma**
This change is intended to start laying the groundwork for generating forms for the redesign:

* Adds JQuery Form Validator to redesign (and tweaks its settings to use proper Bootstrap classes)
* Adds HTML form helpers that are accessible and Bootstrap compliant: `HasErrorFor`, `ShowTextBoxFor`, `ShowValidationMessagesFor`
* Updates the Forgot Password page to use the new HTML form helpers.

### CODE REVIEW

File: src/NuGetGallery/Scripts/gallery/jquery.validate.js [hunk 1]
```diff
@@ -0,0 +1,1231 @@
+﻿/*!
+ * jQuery Validation Plugin 1.11.1
```
Comment (joelverhagen) [inline src/NuGetGallery/Scripts/gallery/jquery.validate.js]: Can we update to latest version? Looks like 1.16.0 is approved.

File: src/NuGetGallery/Scripts/gallery/jquery.validate.unobtrusive.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
+﻿/*!
```
Comment (joelverhagen) [inline src/NuGetGallery/Scripts/gallery/jquery.validate.unobtrusive.js]: Can we take latest?

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 3]
```diff
@@ -186,8 +186,8 @@ private static void BundlingPostStart()
 
             var newScriptBundle = new ScriptBundle("~/Scripts/gallery/site.min.js")
                 .Include("~/Scripts/gallery/jquery-1.12.4.js")
-                .Include("~/Scripts/gallery/jquery.validate.js")
-                .Include("~/Scripts/gallery/jquery.validate.unobtrusive.js")
+                .Include("~/Scripts/gallery/jquery.validate-1.16.0.js")
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/AppActivator.cs]: 👍 👌

**Comment by loic-sharma** [comment]
Updated `jquery.validate.js` and `jquery.validate.unobtrusive`. Both passed [TwC Security Guidance](https://securityguidance.cloudapp.net/) (I had a manual assessment done for `jquery.validate.unobtrusive`).

**Pull request merged**